File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        L ,R = 0, len(numbers) - 1

        while R > L:
            sum_num = numbers[R] + numbers[L]
            if sum_num > target:
                R -= 1
            elif sum_num < target:
                L += 1
            else:
                return [L + 1, R + 1]
                

File: Data Structures & Algorithms/two-integer-sum/submission-1.java

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i=0; i < nums.length; i++) {
            int numberToFind = target - nums[i];
            if(map.containsKey(numberToFind)) {
                return new int[]{map.get(numberToFind), i};
            }

            map.put(nums[i], i);
        }


        return new int[]{};
    }
}
